### DISCUSSION TITLE
Update Ansible Playbook

### DISCUSSION THREAD

**Discussion by adeptvin1** [InterDigitalInc/AdvantEDGE#326] [Show and tell]
Good day!
I does update Ansible Playbook "install-runtime-env.yml" and "install-development-env.yml" and I write "deploy-meepctl.yml".

I checked this playbooks and usally it is to work.
install-runtime-env.yml - always work
install-development-env.yml - often work.
deploy-meepctl.yml - sometimes work, but I think problem in dependences.
I dont understand why, but sometimes meepctl don`t work and I start "export PATH=$PATH:/bin/meepctl" to get start meepctl (without install ./install.sh) 

You can check on http://134.122.105.249/

If you whant I can open issue and add him.

My repo: https://github.com/adeptvin1/AdvantEDGE.git

```
group_vars/all.yml

# Ansible
# ansible_user: root
ansible_python_interpreter: /usr/bin/python3

# Docker
docker_version: "5:19.03"

# Kubernetes
kube_version: "1.19"
cni_version: "0.8.7"
master_ip: "{{ hostvars[groups['master'][0]]['ansible_default_ipv4'].address | default(groups['master'][0]) }}"
network_dir: /etc/kubernetes/network
kubeadmin_config: /etc/kubernetes/admin.conf

# Helm
helm_version: "3.7"

# Go
go_version: "1.17.3"
golangci_lint_version: "v1.43.0"

# Node and NPM
nvm_version: "0.39.0"
node_version: "12.22.7"
npm_version: "6.14.15"
eslint_version: "5.16.0"

# Meepctl
meepctl_lint: "--nolint"
meepctl_node_ip: "{{ hostvars[groups['master'][0]]['ansible_default_ipv4'].address | default(groups['master'][0]) }}"`
```

**Comment by dilallkx** [comment]
Hi @adeptvin1,

If I understand correctly, you are trying to automate the AdvantEDGE deployment procedure using ansible and you are having issues with your playbook execution, correct?

To help identify the issue, could you please confirm that the procedure works when you run the playbook commands manually on your system. Also, I noticed that you upgraded some versions that we may not have tested yet on our end... do you confirm that they work if you run the installation/deployment procedure manually?

Finally, you mention that meepctl does not work in some cases. Are you able to build it? Which meepctl command fails? You can run meepctl with the `-v` option to get verbose log output. Perhaps this can help identify the issue.

I hope this helps identify the cause of your issue. Please let me know when you have more details on the observed issue.
Regards!

**Comment by adeptvin1** [comment]
Hi @dilallkx,
your right, I have some problem with deploy AdvantEDGE in now version Ansible playbook. Because I updated few playbook and add 1 new playbook in my repo. 
I can repeat deploy AdvantEDGE with Ansible and write all log's and confirm work this playbook or I can manually deploy AdvantEDGE.
Now I started deploy AdvantEDGE with Ansible and will send log in this issue.

**Reply by dilallkx** [reply]
Ok, thanks!
If you confirm that deployment works manually with the same versions you are using in your playbook, then it should work. Failures could be due to timing issues during startup (e.g. allowing dependency pods to start before deploying core pods, etc.). The meepctl verbose output (-v) should help identify why certain steps are failing.
I hope this helps you identify the issues.
Regards!
